lib/Runtime/Library/BuiltIn/BuiltIn.js

```diff
@@ -0,0 +1,60 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+"use strict";
+
+(function (intrinsic) {
+    var platform = intrinsic.BuiltIn;
+
+    let FunctionsEnum = {
+        ArrayIndexOf: { className: "Array", methodName: "indexOf", argumentsCount: 1 },
+    };
+
+    platform.registerFunction(FunctionsEnum.ArrayIndexOf, function (searchElement, fromIndex) {
+        let o = __chakraLibrary.Object(this);
```

Comment:
>Object [](start = 32, length = 6)

Is this the Object constructor? The spec defines the Object constructor and the ToObject internal operation differently: 

* [https://tc39.github.io/ecma262/#sec-object-value](https://tc39.github.io/ecma262/#sec-object-value) 
* [https://tc39.github.io/ecma262/#sec-toobject](https://tc39.github.io/ecma262/#sec-toobject)